public/js/NeighbourImages.js

```diff
@@ -83,48 +102,83 @@ NeighbourImages.prototype.getImages = function(sample_id) {
  */
 NeighbourImages.prototype.setImages = function() {
     var self = this;
-    var $nebula0 = $('#nebula-0');
-    var $body = $('body');
-
-    console.log(self.neighbourImages);
 
     // make all img frames empty
-    $nebula0
+    this.selectors.$nebula0
         .attr('src', '//:0')
         .attr('title', 'not found');
 
-    $('.thumbnail > img')
+    this.selectors.$thumbnailChildImg
         .attr('src', '//:0')
         .attr('title', 'not found');
 
     // add selected main image, if it exists
-    if(self.selectedImage) {
-        $nebula0
+    if(this.selectedImage) {
+        this.selectors.$nebula0
             .attr('src', 'data:image/jpg;base64,' +
-                self.selectedImage.image_large)
-            .attr('title', self.selectedImage.sample_id);
+                this.selectedImage.image_large)
+            .attr('title', this.selectedImage.sample_id);
     }
 
-    $nebula0.on('click', function(event) {
-        // prevent browser from scrolling to top when main image clicked on
+    this.selectors.$imgZoomButton.unbind('click');
+    this.selectors.$imgZoomButton.on('click', function(event) {
         event.preventDefault();
+        self.openFullModal();
     });
 
     // iterate through all found images and add them to the gallery
     for(var i = 0; i < self.neighbourImages.length; i++) {
-        var $nebulaSelector = $('#nebula-' + (i+1));
-        $nebulaSelector
+
+        this.selectors.$nebula[i]
             .attr('src', 'data:image/jpg;base64,' +
                 self.neighbourImages[i].image_thumb)
             .attr('title', self.neighbourImages[i].sample_id);
-        $nebulaSelector.unbind('click');
+        this.selectors.$nebula[i].unbind('click');
+
+        // use IIFE here as closure being called in a loop
+        // see http://www.mennovanslooten.nl/blog/post/62
```

Comment:
Immediately Invoked Function Expression. Yes, self calling anonymous functions.. just when you thought it didn't get any better ;)